### PR TITLE
use actions/checkout for grabbing master registry branch instead of manual clone

### PR DESCRIPTION
**Pull request by jrevels**
ref https://github.com/JuliaRegistries/RegistryCI.jl/issues/245

### CODE REVIEW

File: src/AutoMerge/pull-requests.jl [hunk 1]
```diff
@@ -67,42 +70,19 @@ function pull_request_build(pr::GitHub.PullRequest,
...
-                           authorized_authors=authorized_authors,
-                           authorized_authors_special_jll_exceptions=authorized_authors_special_jll_exceptions,
-                           registry_head = registry_head,
-                           registry_master = registry_master,
-                           suggest_onepointzero = suggest_onepointzero,
-                           whoami=whoami,
-                           registry_deps = registry_deps)
-        rm(registry_master; force = true, recursive = true)
+        kind = NewPackage()
+    else is_new_version(pr)
```
Comment (ararslan) [inline src/AutoMerge/pull-requests.jl:74]:
```suggestion
    elseif is_new_version(pr)
```

Comment (ararslan) [reply src/AutoMerge/pull-requests.jl]: Beat me to it :P

**Comment by jrevels** [comment]
CI failures are just because we don't have token/access to the BCBI test repo used by this package's tests

**Comment by ararslan** [comment]
BEAT ME TO IT AGAIN

**Pull request merged**